src/App.js

```diff
@@ -1,24 +1,76 @@
-import React, { Component } from 'react';
-import './App.css';
+import React from "react";
+import Weather from "./components/Weather";
+import Form from "./components/Form";
+import "./App.css";
 
-class App extends Component {
+import Forecast from "./components/Forecast";
 
-  constructor(props) {
-    super(props);
-    this.state = {
-    };
-  }
+const API_KEY = "8d2de98e089f1c28e1a22fc19a24ef04";
+
+class App extends React.Component {
+  state = {
+    temperature: undefined,
+    city: undefined,
+    humidity: undefined,
+    pressure: undefined,
+    hour: undefined,
+    icon: undefined,
+    error: undefined,
+    description: undefined
+  };
+
+  // i use getWeather is a method to make the api call
+  getWeather = async e => {
+    const city = e.target.elements.city.value;
+    e.preventDefault();
+    const api_call = await fetch(
+      `http://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${API_KEY}&units=metric`
+    );
+    const response = await api_call.json();
+    console.log(response);
+    if (city) {
+      this.setState({
+        temperature: response.main.temp,
+        city: response.name,
 
-  render () {
+        humidity: response.main.humidity,
+        pressure: response.main.pressure,
+        hour: response.dt_txt,
+        icon: response.weather[0].icon,
+        description: response.weather[0].description,
+        error: ""
+      });
+    } else {
+      this.setState({
+        error: "Dummy! write city name first"
+      });
+    }
+  };
+
+  render() {
     return (
-      <div className="app">
-        <header className="app__header">
-        </header>
-        <main className="app__main">
-        </main>
+      <div className="flex-container">
+        <div className="main-App">
+          <div className="form-container">
+            <Form getWeather={this.getWeather} />
+          </div>
+
+          <div className="weather-container">
+            <Weather
+              temperature={this.state.temperature}
+              city={this.state.city}
+              humidity={this.state.humidity}
+              pressure={this.state.pressure}
+              hour={this.state.hour}
+              icon={this.state.icon}
+              error={this.state.error}
+              description={this.state.description}
+            />
+          </div>
+        </div>
+        <Forecast />
```

Comment:
So you're already fetching the weather data successfully... can you think of which sections of the data the Forecast component will need, and about how to pass this data down to `Forecast`?

I think it will help if you save _all_ the data (i.e. the whole response) in one piece of state, for example `this.state.weatherData` instead of trying to save little bits of data inside little bits of state (pressure, hour etc)